### PR TITLE
CapSpeedModifier

### PR DESCRIPTION
**Pull request by NikhilSuresh24**
Added a CapSpeedModifier that implements speedmodifier but sets speeds according to an upper and lower bound given to it

### CODE REVIEW

**Comment by leijurv** [comment]
I would suggest using limitValue from Util.Range for consistency (from https://github.com/RoboticsTeam4904/standard/blob/master/Util.java)

**Comment by c0d3rman** [comment]
Agreed, good catch @leijurv

File: subsystems/motor/speedmodifiers/CapSpeedModifier.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package org.usfirst.frc4904.standard.subsystems.motor.speedmodifiers;
+
+
+public class CapSpeedModifier implements SpeedModifier {
+	public final double maxSpeed;
+	public final double minSpeed;
+	
+	public CapSpeedModifier(double max, double min) {
+		maxSpeed = max;
```
Comment (c0d3rman) [inline subsystems/motor/speedmodifiers/CapSpeedModifier.java]: By convention we use the same name for constructor arguments that we do for properties. So this line would look like
```
this.maxSpeed = maxSpeed
```

File: subsystems/motor/speedmodifiers/CapSpeedModifier.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+		minSpeed = min;
+	}
+	
+	@Override
+	public double modify(double speed) {
```
Comment (c0d3rman) [inline subsystems/motor/speedmodifiers/CapSpeedModifier.java]: As @leijurv pointed out, although this is a perfectly fine implementation, for consistency we should make a `Util.Range` in the constructor, and then call `limitValue` on it here.

**Comment by ajnadel** [comment]
@NikhilSuresh24 (@c0d3rman) maybe consider adding a constructor that takes in an existing Util.Range in addition to that one?

File: subsystems/motor/speedmodifiers/CapSpeedModifier.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package org.usfirst.frc4904.standard.subsystems.motor.speedmodifiers;
+
+
+import org.usfirst.frc4904.standard.Util;
+
+public class CapSpeedModifier implements SpeedModifier {
+	public Util.Range speedCap;
+	
+	public CapSpeedModifier(double minSpeed, double maxSpeed) {
+		speedCap = new Util.Range(minSpeed, maxSpeed);
```
Comment (leijurv) [inline subsystems/motor/speedmodifiers/CapSpeedModifier.java]: Perhaps use a self constructor (e.g. replace with `this(new Util.Range(minSpeed, maxSpeed));`)

File: subsystems/motor/speedmodifiers/CapSpeedModifier.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package org.usfirst.frc4904.standard.subsystems.motor.speedmodifiers;
+
+
+import org.usfirst.frc4904.standard.Util;
+
+public class CapSpeedModifier implements SpeedModifier {
+	public Util.Range speedCap;
```
Comment (leijurv) [inline subsystems/motor/speedmodifiers/CapSpeedModifier.java]: Also please make the speedCap either protected or private. If it needs to be accessed, add a getter

**Comment by c0d3rman** [comment]
@leijurv Made the changes you requested

**Comment by leijurv** [comment]
Noticed something while taking another glance, I'm not sure if this is the convention, but you can do `import org.usfirst.frc4904.standard.Util.Range;` and then refer to the class as just `Range` and not  `Util.Range`.

Now it feels like importing `import org.usfirst.frc4904;` and using `standard.Util.Range` everywhere (even though you can't actually do _that_ lol)

**Comment by c0d3rman** [comment]
Since `Range` is a class within `Util` I think this is proper usage

**Comment by leijurv** [comment]
"proper" says who =P

**Comment by c0d3rman** [comment]
Says me

**Comment by leijurv** [comment]
GOTEM

**Comment by ajnadel** [comment]
Leif takes the L

**Comment by leijurv** [comment]
ill take your L

**Pull request merged**